openff/evaluator/layers/equilibration.py

```diff
@@ -0,0 +1,308 @@
+    )
+
+    def validate(self, attribute_type=None):
+        if self.error_tolerances:
+            for error_tolerance in self.error_tolerances:
+                assert isinstance(error_tolerance, EquilibrationProperty)
+                error_tolerance.validate()
+        super(EquilibrationSchema, self).validate(attribute_type)
+
+
+@calculation_layer()
+class EquilibrationLayer(WorkflowCalculationLayer):
+    """A calculation layer which employs molecular simulation
+    to estimate sets of physical properties.
+    """
+
+    @staticmethod
+    def _get_workflow_metadata(
+        working_directory,
+        physical_property,
+        force_field_path,
+        parameter_gradient_keys,
+        storage_backend,
+        calculation_schema,
+    ):
+        """Returns the global metadata to pass to the workflow.
+
+        Parameters
+        ----------
+        working_directory: str
+            The local directory in which to store all local,
+            temporary calculation data from this workflow.
+        physical_property : PhysicalProperty
+            The property that the workflow will estimate.
+        force_field_path : str
+            The path to the force field parameters to use in the workflow.
+        parameter_gradient_keys: list of ParameterGradientKey
+            A list of references to all of the parameters which all observables
+            should be differentiated with respect to.
+        storage_backend: StorageBackend
+            The backend used to store / retrieve data from previous calculations.
+        calculation_schema: EquilibrationLayerSchema
+            The schema containing all of this layers options.
+
+        Returns
+        -------
+        dict of str and Any, optional
+            The global metadata to make available to a workflow.
+            Returns `None` if the required metadata could not be
+            found / assembled.
+        """
+
+        global_metadata = Workflow.generate_default_metadata(
+            physical_property,
+            force_field_path,
+            parameter_gradient_keys,
+            None,  # set target to None for now
+        )
+        global_metadata["error_tolerances"] = copy.deepcopy(
+            calculation_schema.error_tolerances
+        )
+        global_metadata["error_aggregation"] = calculation_schema.error_aggregration
+
+        # search storage for matching boxes already
+        template_queries = calculation_schema.storage_queries
+        for key in template_queries:
+            query = EquilibrationLayer._update_query(
+                template_queries[key],
+                physical_property,
+                calculation_schema,
+            )
+
+            # Apply the query.
+            query_results = storage_backend.query(query)
+
+            objects_to_store = []
+
+            # TODO: should this be hardcoded?
+            # TODO: it is hardcoded in protocols already
+            if key == "component_data":
```

Comment:
See todos -- not sure this should be hardcoded, but it is in the protocols already (and for the existing reweighting layer too)